nixos/modules/services/web-apps/mealie.nix

```diff
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ...}:
+let
+  cfg = config.services.mealie;
+  pkg = cfg.package;
+
+  # TODO Fix ingredient quantities NLP recognition
+in
+{
+  options.services.mealie = {
+    enable = lib.mkEnableOption "Mealie, a recipe manager and meal planner";
+
+    package = lib.mkPackageOption pkgs "mealie" { };
+
+    listenAddress = lib.mkOption {
+      type = lib.types.str;
+      default = "0.0.0.0";
+      description = "Address on which the service should listen.";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9000;
+      description = "Port on which to serve the Mealie service.";
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = with lib.types; attrsOf str;
+      };
```

Comment:
This looks like a flat attribute set, so `attrsOf str` should be sufficient.

Reply:
I tried to apply what was requested here https://github.com/NixOS/nixpkgs/pull/278454#discussion_r1473151732
Isn't it part of the RFC to do so ?

Reply:
The benefit of using the freeformType is, that you can add defaults (e.g. the `port` option in that example) on top of `settings`. if you don't plan on doing that, there is little point in going for this complexity.

Reply:
`attrsOf str` is overly restrictive IMO, given that we `mapAttrs (_: toString)` later.

Reply:
Changed back to `attrsOf anything` so we can use numbers and booleans